numpy/typing/mypy_plugin.py

```diff
@@ -61,13 +61,23 @@ def _get_extended_precision_list() -> t.List[str]:
     return [i.__name__ for i in extended_types if i.__name__ in extended_names]
 
 
+def _get_c_intp_name() -> str:
+    if np.ctypeslib.c_intp is np.intp:
+        return "c_int64"  # Plan B, in case `ctypes` fails to import
+    else:
+        return np.ctypeslib.c_intp.__qualname__
```

Comment:
Something like this ought to be enough:
```suggestion
def _get_c_intp_name() -> str:
    if np.intp is np.int64:
        return "c_int64"
    elif np.intp is np.int32:
        return "c_int32"
    else:
        raise RuntimeError
```

Reply:
Or just repeat these lines:

https://github.com/numpy/numpy/blob/fb215c76967739268de71aa4bda55dd1b062bc2e/numpy/core/_internal.py#L224-L232